src/__tests__/index.js

```diff
@@ -77,34 +79,69 @@ test("customWithCtrl", () => {
     value1: string,
     value2: string
   };
-  const Comp = withCtrl(
-    class extends Component<Props, State> {
-      state = {
-        value1: "state1",
-        value2: "state2"
-      };
-      render() {
-        return <div />;
-      }
-    },
-    {
-      mapPropsToState(props, state) {
-        const overrides = {};
-        if ("value" in props) {
-          const [value1, value2] = props.value.split(" ");
-          overrides.value1 = value1;
-          overrides.value2 = value2;
-        }
-        return overrides;
-      }
+
+  function mapPropsToState(props, state) {
+    const overrides = {};
+    if ("value" in props) {
+      const [value1, value2] = props.value.split(" ");
+      overrides.value1 = value1;
+      overrides.value2 = value2;
+    }
+    return overrides;
+  }
+
+  class Comp extends Component<Props, State> {
+    state = {
+      value1: "state1",
+      value2: "state2"
+    };
+    render() {
+      return (
+        <Ctrl component={this} mapPropsToState={mapPropsToState}>
+          {(state: State) => (
+            <Fragment>{`${state.value1}:${state.value2}`}</Fragment>
+          )}
+        </Ctrl>
+      );
     }
-  );
+  }
 
   const comp1 = mount(<Comp />);
   const comp2 = mount(<Comp defaultValue="default1 default2" />);
   const comp3 = mount(<Comp value="prop1 prop2" />);
 
-  expect(comp1.state()).toEqual({ value1: "state1", value2: "state2" });
-  expect(comp2.state()).toEqual({ value1: "default1", value2: "default2" });
-  expect(comp3.state()).toEqual({ value1: "prop1", value2: "prop2" });
+  expect(comp1.text()).toEqual("state1:state2");
+  expect(comp2.text()).toEqual("default1:default2");
+  expect(comp3.text()).toEqual("prop1:prop2");
+});
+
+test("initial render vs subsequent renders", () => {
+  type Props = {
+    defaultValue?: string,
+    value?: string
+  };
+  type State = {
+    value: string
+  };
+  class Comp extends Component<Props, State> {
+    state = {
+      value: "state"
+    };
+    render() {
+      return (
+        <Ctrl component={this}>
+          {(state: State) => <Fragment>{state.value}</Fragment>}
```

Comment:
Would be brilliant if Flow could infer the type of `state`

Reply:
It would if we injected it directly as proposed above:

```js
<Ctrl state={state} />
```